### PR TITLE
#74 Provide callback for missing fields, #75 provide automatic write support for custom writer, #76 provide automatic read support for custom reader

### PR DESCRIPTION
**Pull request by francisu**
Sorry this pull request contains 3 issues put together, and I imagine it will need some additional work. I do have some code that uses the solutions for issues 75 and 76 that can be added as tests if necessary.

Happy to work with you to improve this.

### CODE REVIEW

**Comment by francisu** [comment]
Also, please ignore the pom changes; they are the ones that I needed to build with.

**Comment by jdereg** [comment]
Thank you for the code submission.  Reviewing.  Should be able to get on this within a few days.

File: src/test/groovy/com/cedarsoftware/util/io/TestMissingFieldHandler.groovy [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        JsonReader.MissingFieldHandler missingHandler = new JsonReader.MissingFieldHandler() {
+            @Override
+            void fieldMissing(Object object, String fieldName, Object value) {
+                ((CustomPoint)object).newY = (long) value;
+            }
+        }
+
+        Map<String,Object> args = [(JsonReader.MISSING_FIELD_HANDLER):missingHandler]
+        CustomPoint clonePoint = JsonReader.jsonToJava(OLD_CUSTOM_POINT,args)
+        assertEquals(pt.x, clonePoint.x)
```
Comment (sgandon) [inline src/test/groovy/com/cedarsoftware/util/io/TestMissingFieldHandler.groovy]: I don't understand why you created *pt* here and not just do 
assertEquals(5, clonePoint.x) ?

File: src/main/java/com/cedarsoftware/util/io/JsonReader.java [hunk 2]
```diff
@@ -57,9 +57,12 @@
     public static final String USE_MAPS = "USE_MAPS";                   // If set, the read-in JSON will be turned into a Map of Maps (JsonObject) representation
     public static final String UNKNOWN_OBJECT = "UNKNOWN_OBJECT";       // What to do when an object is found and 'type' cannot be determined.
     public static final String JSON_READER = "JSON_READER";             // Pointer to 'this' (automatically placed in the Map)
+    public static final String OBJECT_RESOLVER = "OBJECT_RESOLVER";     // Pointer to the current ObjectResolver (automatically placed in the Map)
```
Comment (sgandon) [inline src/main/java/com/cedarsoftware/util/io/JsonReader.java]: I don't see where this object resolver is used in the PR, this seems to be just set and not used.

**Comment by jdereg** [comment]
I am not understanding why you need a missing field handler callback...would you mind explaining further?

Here's what I think you are trying to handle.  You have serialized JSON that has an extra field in it.  You want your missing field handler to be called when it fields a field in the JSON that is not on the object being read into?

If you are using the Map of Maps mode, then this is not an issue as the field will always be placed into the Map.  If you are not using that mode, the field will be dropped.  More information will help here.

For now, I am closing this pull request as it makes some pretty invasive changes (changing the writeObject API from private to public, adding an additional boolean to writeObject 'bodyOnly' - what does that have to do with missing field handling?)

More understanding of what is needed would help.  Also, confining the changes to the least amount possible to support the desired concern would be ideal.

**Pull request closed without merging**